src/main/scala/com/databricks/spark/sql/perf/mllib/data/dataGeneration.scala

```diff
@@ -189,3 +214,48 @@ class GaussianMixtureDataGenerator(
   override def copy(): GaussianMixtureDataGenerator =
     new GaussianMixtureDataGenerator(numCenters, numFeatures, seed)
 }
+
+class RandStringGenerator(
+    distinctCount: Int) extends RandomDataGenerator[String] {
+
+  private val rng = new java.util.Random()
+
+  override def nextValue(): String = {
+    rng.nextInt(distinctCount).toString
+  }
+
+  override def setSeed(seed: Long) {
+    rng.setSeed(seed)
+  }
+
+  override def copy(): RandStringGenerator = new RandStringGenerator(distinctCount)
+}
+
+class DocGenerator(
+    vocabSize: Int,
+    avgDocLength: Int) extends RandomDataGenerator[String] {
+
+  private val wordRng = new java.util.Random()
+  private val numWordsInDocRng = new PoissonGenerator(avgDocLength)
+
+  override def setSeed(seed: Long) {
+    wordRng.setSeed(seed)
+    numWordsInDocRng.setSeed(seed)
+  }
+
+  override def nextValue(): String = {
+    val numWordsInDoc = numWordsInDocRng.nextValue().toInt
```

Comment:
This could technically overflow, so make sure the value is >= 0

Reply:
@jkbradley I use `PoissonGenerator` so it will always >= 0 I think.

Reply:
But can't PoissonGenerator generate values > Int.MAX_VALUE?  It's unlikely but technically possible.